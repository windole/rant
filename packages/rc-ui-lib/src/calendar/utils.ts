/* eslint-disable no-nested-ternary */
export function compareMonth(date1: Date, date2: Date): number {
  const year1 = date1.getFullYear();
  const year2 = date2.getFullYear();

  if (year1 === year2) {
    const month1 = date1.getMonth();
    const month2 = date2.getMonth();
    return month1 === month2 ? 0 : month1 > month2 ? 1 : -1;
  }

  return year1 > year2 ? 1 : -1;
}

export function compareDay(day1: Date, day2: Date): number {
  const compareMonthResult = compareMonth(day1, day2);

  if (compareMonthResult === 0) {
    const date1 = day1.getDate();
    const date2 = day2.getDate();
    return date1 === date2 ? 0 : date1 > date2 ? 1 : -1;
  }

  return compareMonthResult;
}

export const cloneDate = (date: Date): Date => new Date(date);

export const cloneDates = (dates: Date | Date[]): Date | Date[] =>
  Array.isArray(dates) ? dates.map(cloneDate) : cloneDate(dates);

export function getDayByOffset(date: Date, offset: number): Date {
  const cloned = cloneDate(date);
  cloned.setDate(cloned.getDate() + offset);
  return cloned;
}

export const getPrevDay = (date: Date): Date => getDayByOffset(date, -1);
export const getNextDay = (date: Date): Date => getDayByOffset(date, 1);
export const getToday = (): Date => {
  const today = new Date();
  today.setHours(0, 0, 0, 0);
  return today;
};

export function calcDateNum(date: [Date, Date]): number {
  const day1 = date[0].getTime();
  const day2 = date[1].getTime();
  return (day2 - day1) / (1000 * 60 * 60 * 24) + 1;
}

export const getMonthEndDay = (year: number, month: number): number =>
  32 - new Date(year, month - 1, 32).getDate();

const calendarLang = {
  end: '结束',
  start: '开始',
  title: '日期选择',
  confirm: '确定',
  startEnd: '开始/结束',
  weekdays: ['日', '一', '二', '三', '四', '五', '六'],
  monthTitle: (year: number, month: number) => `${year}年${month}月`,
  rangePrompt: (maxRange: number) => `最多选择 ${maxRange} 天`,
};

export const t = (key: string, ...args): string => {
  if (args.length) {
    return calendarLang[key](...args);
  }
  return calendarLang[key];
};

export const formatMonthTitle = (date: Date): string =>
  t('monthTitle', date.getFullYear(), date.getMonth() + 1);
