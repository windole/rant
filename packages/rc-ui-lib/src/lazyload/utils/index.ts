import { ListenEvent } from '../PropsType';

/* eslint-disable prefer-rest-params */
const inBrowser = typeof window !== 'undefined';

export const DEFAULT_URL =
  'data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7';

// 需要监听的事件
export const DEFAULT_EVENTS: ListenEvent[] = [
  'scroll',
  'wheel',
  'resize',
  'mousewheel',
  'animationend',
  'transitionend',
  'touchmove',
];

export const hasIntersectionObserver =
  inBrowser &&
  'IntersectionObserver' in window &&
  'IntersectionObserverEntry' in window &&
  'intersectionRatio' in window.IntersectionObserverEntry.prototype;

export const modeType = {
  event: 'event',
  observer: 'observer',
};

export const call = (fn): void => fn && fn();

export function setImage(
  element: HTMLMediaElement | HTMLIFrameElement | HTMLImageElement | HTMLElement,
  imagePath: string,
): void {
  if (
    element instanceof HTMLIFrameElement ||
    element instanceof HTMLImageElement ||
    element instanceof HTMLMediaElement
  ) {
    element.src = imagePath;
  } else {
    element.style.backgroundImage = `url('${imagePath}')`;
  }
}

export function loadImage(imagePath: string): Promise<string> {
  return new Promise((resolve, reject) => {
    const img = new Image();
    img.src = imagePath;
    img.onload = () => resolve(imagePath);
    img.onerror = (err) => reject(err);
  });
}

export function addCssClassName(
  element: HTMLImageElement | HTMLElement,
  cssClassName: string,
): void {
  if (!element.className.includes(cssClassName)) {
    element.className += ` ${cssClassName}`;
  }
}

export function throttle(fn, threshold = 250, scope?) {
  let last;
  let deferTimer;
  return () => {
    // @ts-ignore
    const context = scope || this;

    const now = +new Date();
    const args = arguments;
    if (last && now < last + threshold) {
      // hold on to it
      clearTimeout(deferTimer);
      deferTimer = setTimeout(() => {
        last = now;
        fn.apply(context, args);
      }, threshold);
    } else {
      last = now;
      fn.apply(context, args);
    }
  };
}

export function debounce(func, wait, immediate?) {
  let timeout;
  let args;
  let context;
  let timestamp;
  let result;

  const later = function later() {
    const last = +new Date() - timestamp;

    if (last < wait && last >= 0) {
      timeout = setTimeout(later, wait - last);
    } else {
      timeout = null;
      if (!immediate) {
        result = func.apply(context, args);
        if (!timeout) {
          context = null;
          args = null;
        }
      }
    }
  };

  return function debounced() {
    // @ts-ignore
    // eslint-disable-next-line @typescript-eslint/no-this-alias
    context = this;
    args = arguments;
    timestamp = +new Date();

    const callNow = immediate && !timeout;
    if (!timeout) {
      timeout = setTimeout(later, wait);
    }

    if (callNow) {
      result = func.apply(context, args);
      context = null;
      args = null;
    }

    return result;
  };
}
