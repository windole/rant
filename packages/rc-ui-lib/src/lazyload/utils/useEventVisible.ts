import { useEffect, useRef, useState } from 'react';
import { debounce as debounceFunc, DEFAULT_EVENTS, throttle as throttleFunc } from '.';
import { EventOptions } from '../PropsType';

// try to handle passive events
let passiveEventSupported = false;
try {
  const opts = Object.defineProperty({}, 'passive', {
    get() {
      passiveEventSupported = true;
    },
  });
  window.addEventListener('test', null, opts);
  // eslint-disable-next-line no-empty
} catch (e) {}

const passiveEvent = passiveEventSupported ? { capture: false, passive: true } : false;

const useEventVisible = (
  eventOptions: EventOptions,
): [React.MutableRefObject<HTMLDivElement | HTMLImageElement>, boolean] => {
  const {
    listenEvents = DEFAULT_EVENTS,
    offset,
    scrollContainer,
    debounce,
    throttle,
  } = eventOptions;
  // 是否可见
  const [visible, setVisible] = useState(false);
  // // 包裹容器
  const containerRef = useRef<HTMLDivElement | HTMLImageElement>(null);

  useEffect(() => {
    // 检查函数
    let checkVisible = () => {
      // 如果容器不存在则不计算
      if (!containerRef.current) return;
      // 获取当前组件位置
      const { top, left, width, height } = containerRef.current?.getBoundingClientRect();
      // 获取滚动容器位置
      const {
        top: parentTop,
        left: parentLeft,
        width: parentWidth,
        height: parentHeight,
      } = scrollContainer.getBoundingClientRect();
      // 计算屏幕高度和宽度
      const windowInnerHeight = window.innerHeight || document.documentElement.clientHeight;
      const windowInnerWidth = window.innerWidth || document.documentElement.clientWidth;
      // 计算滚动容器和屏幕可视区的相交区域
      const intersectionTop = Math.min(parentTop, 0);
      const intersectionLeft = Math.min(parentLeft, 0);

      const intersectionHeight =
        Math.min(windowInnerHeight, parentTop + parentHeight) - intersectionTop;
      const intersectionWidth =
        Math.min(windowInnerWidth, parentLeft + parentWidth) - intersectionLeft;
      // 计算组件距离可视区的高度
      const offsetTop = top - intersectionTop;
      const offsetLeft = left - intersectionLeft;
      // 偏移量计算[top, left]
      const offsets = Array.isArray(offset) ? offset : [offset, offset];
      // 通过上述距离判断组件是否在可见区域
      const isVisible =
        offsetTop - offsets[0] <= intersectionHeight &&
        offsetTop + height + offsets[0] >= 0 &&
        offsetLeft - offsets[1] <= intersectionWidth &&
        offsetLeft + width + offsets[1] >= 0;

      setVisible(isVisible);
      // 如果可见取消当前组件的所有的监听
      if (isVisible) {
        listenEvents.forEach((event) => {
          scrollContainer?.removeEventListener(event, checkVisible);
        });
      }
    };

    // 使用防抖节流增强checkVisible函数，这里优先防抖，其次节流
    if (debounce) {
      // 防抖函数
      checkVisible = debounceFunc(checkVisible, debounce);
    } else if (throttle) {
      // 节流函数
      checkVisible = throttleFunc(checkVisible, throttle);
    }
    // 批量监听事件
    listenEvents.forEach((event) => {
      scrollContainer?.addEventListener(event, checkVisible, passiveEvent);
    });

    // 自动执行一次
    checkVisible();

    return () => {
      listenEvents.forEach((event) => {
        scrollContainer?.removeEventListener(event, checkVisible, passiveEvent);
      });
    };
  }, [scrollContainer]);
  return [containerRef, visible];
};

export default useEventVisible;
