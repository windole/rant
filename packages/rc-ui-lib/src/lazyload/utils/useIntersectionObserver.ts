import { useCallback, useEffect, useState } from 'react';

const useIntersectionObserver = (
  observerOptions: IntersectionObserverInit,
  handleIntersection: IntersectionObserverCallback,
): ((node: Element) => void) => {
  const [element, setElement] = useState<Element>(null);

  const handleElement = useCallback((node: Element) => {
    setElement(node);
  }, []);

  useEffect(() => {
    if (!element) return;
    const observer = new IntersectionObserver(handleIntersection, observerOptions);
    observer.observe(element);
    // eslint-disable-next-line consistent-return
    return () => observer && observer.disconnect && observer.disconnect();
  }, [handleIntersection, observerOptions, element]);
  return handleElement;
};

export default useIntersectionObserver;
