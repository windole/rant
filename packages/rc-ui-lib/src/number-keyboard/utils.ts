import { KeyConfig, NumberKeyboardProps } from './PropsType';

const genBasicKeys = (props: NumberKeyboardProps) => {
  const keys: KeyConfig[] = Array(9)
    .fill('')
    .map((_, i) => ({ text: i + 1 }));

  if (props.randomKeyOrder) {
    keys.sort(() => (Math.random() > 0.5 ? 1 : -1));
  }

  return keys;
};

export const genDefaultKeys = (props: NumberKeyboardProps): KeyConfig[] => [
  ...genBasicKeys(props),
  { text: props.extraKey as string, type: 'extra' },
  { text: 0 },
  {
    text: props.showDeleteKey ? props.deleteButtonText : '',
    type: props.showDeleteKey ? 'delete' : '',
  },
];

export const genCustomKeys = (props: NumberKeyboardProps) => {
  const keys = genBasicKeys(props);
  const { extraKey } = props;
  const extraKeys = Array.isArray(extraKey) ? extraKey : [extraKey];

  if (extraKeys.length === 1) {
    keys.push({ text: 0, wider: true }, { text: extraKeys[0], type: 'extra' });
  } else if (extraKeys.length === 2) {
    keys.push(
      { text: extraKeys[0], type: 'extra' },
      { text: 0 },
      { text: extraKeys[1], type: 'extra' },
    );
  }

  return keys;
};
