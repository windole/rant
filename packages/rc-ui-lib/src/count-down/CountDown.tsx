import React, { useContext, useEffect, useMemo, useImperativeHandle, forwardRef } from 'react';
import classnames from 'classnames';
import { CountDownInstance, CountDownProps } from './PropsType';
import ConfigProviderContext from '../config-provider/ConfigProviderContext';
import { parseFormat } from './utils';
import { CurrentTime, useCountDown } from '../hooks/use-count-down';

const CountDown = forwardRef<CountDownInstance, CountDownProps>((props, ref) => {
  const { time, format, autoStart, millisecond, renderChildren } = props;

  const { start, pause, reset, current } = useCountDown({
    time: +time,
    millisecond,
    onChange: (value: CurrentTime) => props.onChange?.(value),
    onFinish: props.onFinish,
  });
  const timeText = useMemo(() => parseFormat(format, current), [format, current]);

  const resetTime = () => {
    reset(+time);

    if (autoStart) {
      start();
    }
  };

  useEffect(() => {
    resetTime();
  }, [time]);

  useImperativeHandle(ref, () => ({
    start,
    pause,
    reset: resetTime,
  }));

  const { prefixCls, createNamespace } = useContext(ConfigProviderContext);
  const [bem] = createNamespace('count-down', prefixCls);

  return (
    <div role="timer" className={classnames(bem())}>
      {renderChildren ? renderChildren(current) : timeText}
    </div>
  );
});

CountDown.defaultProps = {
  time: 0,
  format: 'HH:mm:ss',
  autoStart: true,
  millisecond: false,
};

export default CountDown;
