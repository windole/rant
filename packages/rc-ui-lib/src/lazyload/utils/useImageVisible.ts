import { useCallback, useState } from 'react';
import { addCssClassName, loadImage, setImage } from '.';
import { IntersectionObserverEntryType, ObserverOptions } from '../PropsType';
import useIntersectionObserver from './useIntersectionObserver';

const defaultObserverOptions = {
  threshold: [0, 1],
};

const useImageVisible = (
  observerOptions: ObserverOptions = defaultObserverOptions,
  errorImage: string,
): [(node: Element) => void, boolean] => {
  const [isLoaded, setIsLoaded] = useState(false);

  const handleIntersection = useCallback(
    (entries: IntersectionObserverEntryType[]) => {
      entries
        .filter((entry) => entry.isIntersecting)
        .forEach((entry) => {
          const target = entry.target as HTMLImageElement | HTMLElement;
          const url = target.dataset.src || '';
          loadImage(url)
            .catch(() => {
              setImage(target, errorImage);
              addCssClassName(target, 'rc-lazyload__error');
              target.setAttribute('lazy', 'error');
            })
            .then((imagePath: string) => {
              setImage(target, imagePath);
              setIsLoaded(true);
              addCssClassName(target, 'rc-lazyload__loaded');
              target.setAttribute('lazy', 'loaded');
              target.removeAttribute('data-src');
            });
        });
    },
    [setIsLoaded, isLoaded],
  );
  const ref = useIntersectionObserver(observerOptions, handleIntersection);
  return [ref, isLoaded];
};

export default useImageVisible;
