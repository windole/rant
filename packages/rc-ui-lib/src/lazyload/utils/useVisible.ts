import { useCallback, useEffect, useState } from 'react';
import { addCssClassName } from '.';
import useIntersectionObserver from './useIntersectionObserver';

const defaultObserverOptions = {
  threshold: [0, 1],
};

const useVisible = (
  initialVisible: boolean,
  observerOptions = defaultObserverOptions,
): [(node: Element) => void, boolean] => {
  const [visible, setVisible] = useState(initialVisible);

  const forceVisible = () => {
    setVisible(true);
  };

  useEffect(() => {
    if (initialVisible) {
      forceVisible();
    }
  }, [initialVisible]);

  const handleIntersection = useCallback(
    (entries) => {
      entries
        .filter((entry) => entry.isIntersecting)
        .forEach((entry) => {
          const target = entry.target as HTMLElement;
          addCssClassName(target, 'rc-lazyload__loaded');
          setVisible(true);
        });
    },
    [setVisible, visible],
  );
  const ref = useIntersectionObserver(observerOptions, handleIntersection);
  return [ref, visible];
};

export default useVisible;
