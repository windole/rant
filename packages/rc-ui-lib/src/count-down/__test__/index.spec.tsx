/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render, cleanup, waitFor } from '@testing-library/react';
import { sleep } from '../../../tests/utils';
import CountDown, { CountDownInstance, CountDownProps } from '..';

describe('CountDown test with testing library', () => {
  function createCountDown(props: CountDownProps) {
    const countDownRef = React.createRef<CountDownInstance>();

    const { queryByTestId, container, rerender, debug, baseElement } = render(
      <CountDown ref={countDownRef} {...props} />,
    );

    return {
      container,
      rerender,
      countDownRef,
      debug,
      queryByTestId,
      baseElement,
    };
  }

  beforeEach(() => {
    jest.useFakeTimers();
  });

  afterEach(() => {
    cleanup();
    jest.restoreAllMocks();
  });

  it('should emit finish event when finished', async () => {
    Object.defineProperty(window, 'inBrowser', { value: true });
    const onFinish = jest.fn();

    createCountDown({
      time: 1,
      onFinish,
    });

    expect(onFinish).not.toHaveBeenCalled();

    await sleep(100);

    expect(onFinish).toHaveBeenCalled();
  });

  it('should emit finish event when finished and millisecond is true', async () => {
    Object.defineProperty(window, 'inBrowser', { value: true });
    const onFinish = jest.fn();

    createCountDown({
      time: 1,
      millisecond: true,
      onFinish,
    });

    expect(onFinish).not.toHaveBeenCalled();

    await sleep(100);

    expect(onFinish).toHaveBeenCalled();
  });

  it('should re-render after some time', async () => {
    Object.defineProperty(window, 'inBrowser', { value: true });

    const { container } = createCountDown({
      time: 1000,
      format: 'SSS',
    });
    const preSnapshot = container.querySelector('.rc-count-down').innerHTML;

    await sleep(100);
    const laterSnapshot = container.querySelector('.rc-count-down').innerHTML;

    expect(preSnapshot).not.toEqual(laterSnapshot);
  });

  it('should not start counting when auto-start prop is false', async () => {
    Object.defineProperty(window, 'inBrowser', { value: true });

    const { container } = createCountDown({
      time: 100,
      format: 'SSS',
      autoStart: false,
    });

    await sleep(50);

    expect(container).toMatchSnapshot();
  });
  it('should start counting after calling the start method', async () => {
    Object.defineProperty(window, 'inBrowser', { value: true });

    const { countDownRef, container } = createCountDown({
      time: 100,
      format: 'SSS',
      autoStart: false,
      millisecond: true,
    });

    const preSnapshot = container.querySelector('.rc-count-down').innerHTML;

    await waitFor(async () => {
      countDownRef.current.start();
    });
    await sleep(50);
    const laternapshot = container.querySelector('.rc-count-down').innerHTML;

    expect(preSnapshot).not.toEqual(laternapshot);
  });
  it('should pause counting after calling the pause method', async () => {
    Object.defineProperty(window, 'inBrowser', { value: true });

    const { countDownRef, container } = createCountDown({
      time: 100,
      format: 'SSS',
      millisecond: true,
    });
    expect(container).toMatchSnapshot();

    await waitFor(async () => {
      countDownRef.current.pause();
    });

    await sleep(50);

    expect(container).toMatchSnapshot();
  });
  it('should reset time after calling the reset method', async () => {
    Object.defineProperty(window, 'inBrowser', { value: true });

    const { countDownRef, container } = createCountDown({
      time: 100,
      format: 'SSS',
      autoStart: false,
      millisecond: true,
    });
    expect(container).toMatchSnapshot();

    await waitFor(async () => {
      countDownRef.current.start();
    });

    await sleep(50);

    await waitFor(async () => {
      countDownRef.current.reset();
    });

    expect(container).toMatchSnapshot();
  });
  it('should format complete time correctly', async () => {
    Object.defineProperty(window, 'inBrowser', { value: true });

    const { container } = createCountDown({
      time: 30 * 60 * 60 * 1000 - 1,
      format: 'DD-HH-mm-ss-SSS',
      autoStart: false,
      millisecond: true,
    });

    expect(container).toMatchSnapshot();
  });
  it('should format incomplete time correctly', async () => {
    Object.defineProperty(window, 'inBrowser', { value: true });

    const { container } = createCountDown({
      time: 30 * 60 * 60 * 1000 - 1,
      format: 'HH-mm-ss-SSS',
      autoStart: false,
      millisecond: true,
    });

    expect(container).toMatchSnapshot();
  });
  it('should format SS milliseconds correctly', async () => {
    Object.defineProperty(window, 'inBrowser', { value: true });

    const { container } = createCountDown({
      time: 1500,
      format: 'ss-SS',
      autoStart: false,
      millisecond: true,
    });

    expect(container).toMatchSnapshot();
  });
  it('should format S milliseconds correctly', async () => {
    Object.defineProperty(window, 'inBrowser', { value: true });

    const { container } = createCountDown({
      time: 1500,
      format: 'ss-S',
      autoStart: false,
      millisecond: true,
    });

    expect(container).toMatchSnapshot();
  });
  it('should render correctly using renderChilden', async () => {
    Object.defineProperty(window, 'inBrowser', { value: true });

    const { container } = createCountDown({
      time: 1500,
      autoStart: false,
      renderChildren(timeData) {
        return (
          <>
            <span className="block">{timeData.hours}</span>
            <span className="colon">:</span>
            <span className="block">{timeData.minutes}</span>
            <span className="colon">:</span>
            <span className="block">{timeData.seconds}</span>
          </>
        );
      },
    });

    expect(container).toMatchSnapshot();
  });

  it('should emit change event when counting', async () => {
    Object.defineProperty(window, 'inBrowser', { value: true });
    const onChange = jest.fn();

    const { countDownRef } = createCountDown({
      time: 1,
      autoStart: false,
      onChange,
    });

    await waitFor(async () => {
      countDownRef.current.start();
    });

    await sleep(100);

    expect(onChange).toHaveBeenCalledWith({
      days: 0,
      hours: 0,
      milliseconds: 0,
      minutes: 0,
      seconds: 0,
      total: 0,
    });
  });
});
