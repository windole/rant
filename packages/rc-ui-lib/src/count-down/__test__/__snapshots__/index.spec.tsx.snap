// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountDown test with testing library should format S milliseconds correctly 1`] = `
<div>
  <div
    class="rc-count-down"
    role="timer"
  >
    01-5
  </div>
</div>
`;

exports[`CountDown test with testing library should format SS milliseconds correctly 1`] = `
<div>
  <div
    class="rc-count-down"
    role="timer"
  >
    01-50
  </div>
</div>
`;

exports[`CountDown test with testing library should format complete time correctly 1`] = `
<div>
  <div
    class="rc-count-down"
    role="timer"
  >
    01-05-59-59-999
  </div>
</div>
`;

exports[`CountDown test with testing library should format incomplete time correctly 1`] = `
<div>
  <div
    class="rc-count-down"
    role="timer"
  >
    29-59-59-999
  </div>
</div>
`;

exports[`CountDown test with testing library should not start counting when auto-start prop is false 1`] = `
<div>
  <div
    class="rc-count-down"
    role="timer"
  >
    100
  </div>
</div>
`;

exports[`CountDown test with testing library should pause counting after calling the pause method 1`] = `
<div>
  <div
    class="rc-count-down"
    role="timer"
  >
    100
  </div>
</div>
`;

exports[`CountDown test with testing library should pause counting after calling the pause method 2`] = `
<div>
  <div
    class="rc-count-down"
    role="timer"
  >
    100
  </div>
</div>
`;

exports[`CountDown test with testing library should render correctly using renderChilden 1`] = `
<div>
  <div
    class="rc-count-down"
    role="timer"
  >
    <span
      class="block"
    >
      0
    </span>
    <span
      class="colon"
    >
      :
    </span>
    <span
      class="block"
    >
      0
    </span>
    <span
      class="colon"
    >
      :
    </span>
    <span
      class="block"
    >
      1
    </span>
  </div>
</div>
`;

exports[`CountDown test with testing library should reset time after calling the reset method 1`] = `
<div>
  <div
    class="rc-count-down"
    role="timer"
  >
    100
  </div>
</div>
`;

exports[`CountDown test with testing library should reset time after calling the reset method 2`] = `
<div>
  <div
    class="rc-count-down"
    role="timer"
  >
    100
  </div>
</div>
`;
